### PR TITLE
Feature/iat 3181

### PR DESCRIPTION
**Pull request by bdrainer**
The event handler was previously in common.  History is out of common.

### CODE REVIEW

File: src/main/java/org/opentestsystem/ap/item/history/ItemHistoryEventHandler.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        );
+    }
+
+    /**
+     * The item is used to generated SAAIF files in the given item folder.
+     *
+     * @param item       The item to generate SAAIF for.
+     * @param itemFolder The folder to generate the data in.
+     */
+    private void generateSaaifContent(Item item, Path itemFolder) {
```
Comment (millerts) [inline src/main/java/org/opentestsystem/ap/item/history/ItemHistoryEventHandler.java]: This isn't something that needs to be handled here, but it seems like the logic in this method is spread throughout TIMS.  In the future we may want to consolidate this into a single call in a common Manager that everyone uses.

File: src/main/java/org/opentestsystem/ap/item/history/ItemHistoryEventHandler.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+     * @param attachmentVersionId The version of the attachments to download.
+     * @return The path were the attachments have been downloaded.
+     */
+    private Path downloadAttachments(String itemId, String attachmentVersionId) {
+        // use 'localBaseDir' so the clean up process takes care of deleting these files
+        Path downloadFolder = Paths.get(itemBankProperties.getLocalBaseDir(), idGenerator.generateTransactionId());
+        if (StringUtils.isNotBlank(attachmentVersionId)) {
+            dataStoreAttachmentManager.downloadVersionedAttachments(itemId, attachmentVersionId, downloadFolder);
+        } else {
+            dataStoreAttachmentManager.downloadAttachments(itemId, BRANCH_MASTER, downloadFolder);
```
Comment (millerts) [inline src/main/java/org/opentestsystem/ap/item/history/ItemHistoryEventHandler.java]: Should there a backup or should this outright fail?  My concern is that if the attachment version id is required we shouldn't silently use master?

File: src/main/java/org/opentestsystem/ap/item/history/ItemHistoryEventHandler.java (hunk 1 above)
Comment (bdrainer) [reply src/main/java/org/opentestsystem/ap/item/history/ItemHistoryEventHandler.java]: That's a good idea.

File: src/main/java/org/opentestsystem/ap/item/history/ItemHistoryEventHandler.java (hunk 2 above)
Comment (bdrainer) [reply src/main/java/org/opentestsystem/ap/item/history/ItemHistoryEventHandler.java]: Let's fail.  When committing changes on an item attachment version ID is required.  If it isn't present something is wrong.  You are right silently using master is a bad idea.

Let's discuss this tomorrow at parking lot.  I want to describe how an item **commit** event produces a new attachment version but an item **patch** event does not, it just reuses what is already set on the item if anything at all.  Maybe commit and patch are handled differently by the history service where I have those two events handled identically.

File: src/main/java/org/opentestsystem/ap/item/history/ItemHistoryEventHandler.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+     * attachments are downloaded individually.  This is a fallback behavior as it is expected the version ID is
+     * provided as it should have been generate prior to the item event being sent.
+     *
+     * @param itemId              The item to download the attachments for.
+     * @param attachmentVersionId The version of the attachments to download.
+     * @return The path were the attachments have been downloaded.
+     */
+    private Path downloadAttachments(String itemId, String attachmentVersionId) {
+        if (StringUtils.isBlank(attachmentVersionId)) {
+            throw new SystemException("[TIMS-HISTORY-ERROR] Item " + itemId + " is missing attachment version ID.");
```
Comment (bdrainer) [inline src/main/java/org/opentestsystem/ap/item/history/ItemHistoryEventHandler.java]: Here is where we now throw when the attachment version ID is blank.  I put the TIMS-HISTORY-ERROR in the error message.  https://smarterbalanced.atlassian.net/wiki/spaces/IAT/pages/22020097/TIMS+System+Monitoring - this page has been updated with the new error key to monitor.

**Pull request merged**